test/export/testing.py

```diff
@@ -0,0 +1,43 @@
+import functools
+import unittest
+from unittest.mock import patch
+
+
+def make_test_cls_with_mocked_export(
+    cls, cls_prefix, fn_suffix, mocked_export_fn, xfail_prop=None
+):
+    MockedTestClass = type(f"{cls_prefix}{cls.__name__}", cls.__bases__, {})
+    MockedTestClass.__qualname__ = MockedTestClass.__name__
+
+    for name in dir(cls):
+        if name.startswith("test_"):
+            fn = getattr(cls, name)
+            if not callable(fn):
+                setattr(MockedTestClass, name, getattr(cls, name))
+                continue
+            new_name = f"{name}{fn_suffix}"
+            new_fn = _make_fn_with_mocked_export(fn, mocked_export_fn)
+            new_fn.__name__ = new_name
+            if xfail_prop is not None and hasattr(fn, xfail_prop):
+                new_fn = unittest.expectedFailure(new_fn)
+            setattr(MockedTestClass, new_name, new_fn)
+        # NB: Doesn't handle slots correctly, but whatever
+        elif not hasattr(MockedTestClass, name):
+            setattr(MockedTestClass, name, getattr(cls, name))
+
+    return MockedTestClass
+
+
+def _make_fn_with_mocked_export(fn, mocked_export_fn):
+    @functools.wraps(fn)
+    def _fn(*args, **kwargs):
+        with patch("test_export.export", mocked_export_fn):
```

Comment:
That's on purpose. Here we keep the original handler `torch.export.export` and only change the behavior of `test_export.export`. It won't break as long as test_export.py has `from torch.export import export`.

Reply:
Why don't we just patch the original handler? Since this is within this context manager, i feel like it should be fine for whoever is using export outside of this file

Reply:
The reason is we could introduce infinite recursion if we patch directly on the real export API. In the way here, we only patch the surfaced export in the test file. Does it make sense?

Reply:
Yeah makes sense. Can you add a comment on test_export.py that we shouldn't change the import statement?